presto-main/src/test/java/com/facebook/presto/operator/TestScanFilterAndProjectOperator.java

```diff
@@ -181,6 +183,66 @@ public void testRecordCursorSource()
     }
 
     @Test
+    public void testPageYield()
+            throws Exception
+    {
+        int totalRows = 1000;
+        Page input = SequencePageBuilder.createSequencePage(ImmutableList.of(BIGINT), totalRows, 1);
+        DriverContext driverContext = newDriverContext();
+
+        // 20 columns; each column is associated with a function that will force yield per projection
+        int totalColumns = 20;
+        ImmutableList.Builder<SqlScalarFunction> functions = ImmutableList.builder();
+        for (int i = 0; i < totalColumns; i++) {
+            functions.add(new GenericLongFunction("page_col" + i, value -> {
+                driverContext.getYieldSignal().forceYieldForTesting();
+                return value;
+            }));
+        }
+        Metadata metadata = functionAssertions.getMetadata();
+        metadata.getFunctionRegistry().addFunctions(functions.build());
+
+        // match each column with a projection
+        ExpressionCompiler expressionCompiler = new ExpressionCompiler(metadata, new PageFunctionCompiler(metadata, 0));
+        ImmutableList.Builder<RowExpression> projections = ImmutableList.builder();
+        for (int i = 0; i < totalColumns; i++) {
+            projections.add(call(internalScalarFunction("generic_long_page_col" + i, BIGINT.getTypeSignature(), ImmutableList.of(BIGINT.getTypeSignature())), BIGINT, field(0, BIGINT)));
+        }
+        Supplier<CursorProcessor> cursorProcessor = expressionCompiler.compileCursorProcessor(Optional.empty(), projections.build(), "key");
+        Supplier<PageProcessor> pageProcessor = expressionCompiler.compilePageProcessor(Optional.empty(), projections.build());
+
+        ScanFilterAndProjectOperator.ScanFilterAndProjectOperatorFactory factory = new ScanFilterAndProjectOperator.ScanFilterAndProjectOperatorFactory(
+                0,
+                new PlanNodeId("test"),
+                new PlanNodeId("0"),
+                (session, split, columns) -> new FixedPageSource(ImmutableList.of(input)),
+                cursorProcessor,
+                pageProcessor,
+                ImmutableList.of(),
+                ImmutableList.of(BIGINT));
+
+        SourceOperator operator = factory.createOperator(driverContext);
+        operator.addSplit(new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit()));
+        operator.noMoreSplits();
+
+        // we only check yield signal once a column has been completely processed; thus the first 19 outputs will get nothing
+        for (int i = 0; i < totalColumns - 1; i++) {
+            driverContext.getYieldSignal().setWithDelay(5 * SECONDS.toNanos(1), driverContext.getYieldExecutor());
```

Comment:
I saw this in `THJO.testYield` and found it misleading (and preventing any form of debugging). Wouldn't it be sufficient to let your `GenericLongFunction` force yield?